### ISSUE TITLE
Support Protected branches

### ISSUE THREAD

**Issue by roidelapluie** [gitlabhq/terraform-provider-gitlab#17]

**Comment by roidelapluie** [comment]
Blocked by upstream https://github.com/xanzy/go-gitlab/issues/216

**Comment by nextrevision** [comment]
That issue has since been resolved. This should be unblocked.

**Comment by grv87** [comment]
This issue is resolved in 8da2712eda93bd0ceeebbcc693ffccd5a7444279 and subsequent commits, but still missing docs, isn't it?

**Comment by VMitov** [comment]
I just tested the undocumented `gitlab_branch_protection` resource but there is one issue with it. It doesn't force the state of the resource if it is already in the state but changed manually somehow. If you don't have the protected branches in the state - it works as expected.

**Comment by md5** [comment]
@VMitov I just came across the same problem. Manual changes to the branch protection were not detected by a `terraform refresh` and a `terraform apply` did not put the resource into the specified state. Seems like that deserves its own issue perhaps?

**Comment by md5** [comment]
Also, this `gitlab_branch_protection` resource is no longer undocumented: https://www.terraform.io/docs/providers/gitlab/r/branch_protection.html

**Comment by armsnyder** [comment]
Fixed via #166
